lib/core/sync_job_runner.rb

```diff
@@ -59,11 +56,17 @@ def do_sync!
 
         Utility::Logger.debug("#{existing_ids.size} documents are present in index #{@connector_settings.index_name}.")
 
-        @connector_instance.yield_documents do |document|
+        reporting_cycle_start = Time.now
+        @connector_instance.yield_documents do |document, connector_metadata = {}|
           document = add_ingest_metadata(document)
           @sink.ingest(document)
           incoming_ids << document['id']
-          @status[:indexed_document_count] += 1
+          @connector_metadata = connector_metadata
```

Comment:
This re-assigns an instance variable every time a single document is emitted. That feels odd to me, that we aren't aggregating that metadata, but totally re-assigning it each time. If we wanted it fresh, would it not make more sense to add a `def metadata` function to the connector, so that we could just call `@connector_instance.metadata` whenever we wanted, rather than making the connector `yield` it with the documents?

Reply:
Great suggestion @seanstory . Will do so.